### PR TITLE
Add Mongo unit test classes for latest 3.6.20 and 4.4.1

### PR DESCRIPTION
**Pull request by adamjshook**
Relates to #5804

This commit refactors the distributed query test into a base with three implementations, the existing unit tests against the original 3.4.0, then two classes against the latest in the 3.x line, 3.6.20, and the latest in 4.x, 4.4.1.

### CODE REVIEW

File: presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/BaseMongoDistributedQueries.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
+/*
```
Comment (kokosing) [inline presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/BaseMongoDistributedQueries.java:1]: Mention supported versions to docs.

File: presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/BaseMongoDistributedQueries.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import io.prestosql.testing.sql.TestTable;
+import org.testng.SkipException;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public abstract class BaseMongoDistributedQueries
```
Comment (kokosing) [inline presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/BaseMongoDistributedQueries.java:26]: Please extract refactor to separate commit.

File: presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/BaseMongoDistributedQueries.java (hunk 1 above)
Comment (adamjshook) [reply presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/BaseMongoDistributedQueries.java:1]: Put them in a separate commit, let me know if I should roll them in to the second commit.

File: presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/BaseMongoDistributedQueries.java [hunk 3]
```diff
@@ -13,33 +13,21 @@
  */
```
Comment (kokosing) [inline presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/BaseMongoDistributedQueries.java:13]: I would name this commit as: `Extract BaseMongoDistributedQueries`. Also please that we are removing test coverage here and restoring in subsuequent commit, and that is because that we won't to retain the file changes history.

File: presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/BaseMongoDistributedQueries.java [hunk 4]
```diff
@@ -13,33 +13,21 @@
...
 
 @Test
-public class TestMongoDistributedQueries
+public abstract class BaseMongoDistributedQueries
         extends AbstractTestDistributedQueries
 {
-    private MongoServer server;
-
-    @Override
-    protected QueryRunner createQueryRunner()
```
Comment (kokosing) [inline presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/BaseMongoDistributedQueries.java]: Maybe leave this method and instead provide `createMongoServer`?

Another option (preferable by me) would be to remove `createQueryRunner` method and `server` field, and maintain it in the class that creates it. More code is duplicated, but I find inheritance of infrastructure as bad practice.

File: presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/TestMongoDistributedQueriesLatest.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import static io.prestosql.plugin.mongodb.MongoQueryRunner.createMongoQueryRunner;
+
+public class TestMongoDistributedQueriesLatest
+        extends BaseMongoDistributedQueries
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        MongoServer server = closeAfterClass(new MongoServer("mongo:3.6.20"));
```
Comment (kokosing) [inline presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/TestMongoDistributedQueriesLatest.java:29]: Why latest is 3, and not 4?

Comment (adamjshook) [reply presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/TestMongoDistributedQueriesLatest.java:29]: We are testing the latest 3.x here and there is a `TestMongo4DistributedQueriesLatest` for the latest in major version 4.

File: presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/MongoServer.java [hunk 6]
```diff
@@ -27,7 +27,12 @@
 
     public MongoServer()
     {
-        this.dockerContainer = new MongoDBContainer("mongo:3.4.0")
+        this("mongo:3.4.0");
+    }
+
+    public MongoServer(String dockerImageName)
```
Comment (kokosing) [inline presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/MongoServer.java:33]: The best would be to pass just a version, not entire docker image name.

File: presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/TestMongoDistributedQueriesLatest.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+ */
+package io.prestosql.plugin.mongodb;
+
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.testing.QueryRunner;
+import io.prestosql.tpch.TpchTable;
+
+import static io.prestosql.plugin.mongodb.MongoQueryRunner.createMongoQueryRunner;
+
+public class TestMongoDistributedQueriesLatest
```
Comment (kokosing) [inline presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/TestMongoDistributedQueriesLatest.java:22]: Assuming that we have two latests classes, let's call it `TestMongo3DistributedQueriesLatest`.

**Review by kokosing — APPROVED** [review]
% two minor comments

**Review by kokosing — APPROVED** [review]

**Comment by kokosing** [comment]
Merged, thanks!
